lib/credentials/AccountCredentials.js

```diff
@@ -2,14 +2,20 @@ const assert = require('assert');
 const AWS = require('aws-sdk');
 
 const errors = require('arsenal').errors;
-const management = require('../../../lib/management');
+const management = require('../../lib/management');
+
+const authdata = require('../../conf/authdata.json');
 
 class AccountCredentials extends AWS.Credentials {
-    constructor(authConfig, log) {
-        this._log = log;
-        this.initAuthData();
-        const accountInfo = this.getAuthData().accounts.find(
-            account => account.name === authConfig.account);
+    constructor(authConfig, log, serviceAuth) {
```

Comment:
Why move the provisioned service account logic here? it makes `ProvisionedServiceAccountCredentials` basically a passthrough while the base class now has knowledge of several auth types